### ISSUE TITLE
Ignore user specified dirs

### ISSUE THREAD

**Issue by jirpok** [SeeLog/python-init-generator#16]
Hey,

thanks for the extension. It's a minor issue, but would it be possible to implement a mechanism to allow users to ignore certain directories? Unless I'm missing sth.

Best, 

Jiri

**Comment by sisoe24** [comment]
Hi! Currently, there isn't a way to exclude directories while generating init files. I usually use the right-click "Generate __init__.py" option, but it will still recursively go into every subfolder that contains a .py file and place one in it. 

It would be useful to have a feature that allows the exclusion of directories, and it shouldn't be too difficult to implement. We'll need to wait to see what @SeeLog thinks.

**Comment by still-dreaming-1** [comment]
Hello. I wonder if it would be better to do this the other way around, perhaps having an option for directories to include, or some convention to know what to expect for what it will include.

Given there is not currently a way to exclude (or include) specific directories, in the case that we were to run the more general command from the command palette (so not right clicking a directory from the Explorer), how does it decide which directories to create `__init__.py` files for? I tried running it, and it did exactly what I want, but I still would like to know so I don't have unexpected behavior later.

I may end up running this command automatically at some point, so I want to get it setup to always create the files in the right places, and not create any in the wrong places. My project structure is pretty standard. There is the root project directory, and then all the source code is in a directory (and subdirectories of it) that has the same name as the project. So basically there is a single top-level package directory with subpackages in the subdirectories. So let's say my top-level package is called "pybot", I would only want this tool to look in the "pybot" directory and it's subdirectories for potential places to create.

Perhaps a good default behavior would be to look inside the pyproject.toml file, and if there is has something like
```
[project]
name = "pybot"
```

It would look for a `pybot` directory inside the root project directory, and only consider making `__init__.py` files there and in its subdirectories. Or maybe that is how it already works?

**Comment by sisoe24** [comment]
Hi. I made a PR available with the ignore directories feature. The ignore directories can be used as a regex pattern to have more control over the list. Since Seelog is not currently present, you might have to pull the code and install the package locally to try the new features. Otherwise, if requested, I can supply the `extension.vsix` file for manual installation. Let me know.

**Comment by still-dreaming-1** [comment]
This is a little off topic, but I wanted to point this out here in case anyone else wants to automate the generation of these files even further.

I ended up just making my own python script that creates these files for me. I also setup a VSCode task that runs it. I have the task configured to run automatically in response to certain [File Watcher](https://marketplace.visualstudio.com/items?itemName=appulate.filewatcher) events. Now I don't need to even think about running a command or task to generate these files, it just happens automatically. Of course that might not always be a good thing depending on your needs or your project, but I like having it in place for the project I'm working on.

Here are my File Watcher settings. `gen-inits` is just the name of the VSCode task I made that generates the files.

```
    "filewatcher.commands": [
      {
        "match": "\\.py*",
        "isAsync": true,
        "vscodeTask": ["workbench.action.tasks.runTask", "gen-inits"],
        "event": "onFileCreate",
      }, {
        "match": "\\.py*",
        "isAsync": true,
        "vscodeTask": ["workbench.action.tasks.runTask", "gen-inits"],
        "event": "onFileRename",
      },
    ],
```

**Comment by sisoe24** [comment]
> Hello. I wonder if it would be better to do this the other way around, perhaps having an option for directories to include, or some convention to know what to expect for what it will include.
> 
> Given there is not currently a way to exclude (or include) specific directories, in the case that we were to run the more general command from the command palette (so not right clicking a directory from the Explorer), how does it decide which directories to create `__init__.py` files for? I tried running it, and it did exactly what I want, but I still would like to know so I don't have unexpected behavior later.
> 
> I may end up running this command automatically at some point, so I want to get it setup to always create the files in the right places, and not create any in the wrong places. My project structure is pretty standard. There is the root project directory, and then all the source code is in a directory (and subdirectories of it) that has the same name as the project. So basically there is a single top-level package directory with subpackages in the subdirectories. So let's say my top-level package is called "pybot", I would only want this tool to look in the "pybot" directory and it's subdirectories for potential places to create.
> 
> Perhaps a good default behavior would be to look inside the pyproject.toml file, and if there is has something like
> 
> ```
> [project]
> name = "pybot"
> ```
> 
> It would look for a `pybot` directory inside the root project directory, and only consider making `__init__.py` files there and in its subdirectories. Or maybe that is how it already works?

I'm sorry for not replying earlier, but I'm glad you were able to figure it out. The extension creates an `__init__.py` file in every directory that contains a `*.py` file. The root directory where the file is created depends on which command you use. If you use the context menu, it will only create the file from the selected folder. However, using the command palette command will create the file in all the directories in your workspace. With the new PR, you can exclude certain directories using a regex pattern.

**Comment by SeeLog** [comment]
Sorry for late reply.
I think this issue should be resolved by #17.

I will release newer version.
Thank you so much @sisoe24
